drivers/lora/sx12xx_common.c

```diff
@@ -80,14 +94,25 @@ int sx12xx_lora_recv(const struct device *dev, uint8_t *data, uint8_t size,
 {
 	int ret;
 
+	/* Ensure available, decremented by sx12xx_ev_rx_done or on timeout */
+	if (!MODEM_CLAIM(dev_data)) {
+		return -EBUSY;
+	}
+
+	/* Clear any previously triggered packets */
+	k_sem_take(&dev_data.data_sem, K_NO_WAIT);
```

Comment:
This looks a bit suspicious to me. If we can end up in a situation where the semaphore is non-zero here, wouldn't  k_sem_reset be a better option?

Reply:
This is not entirely correct, given that the semaphore is declared with `K_SEM_MAX_LIMIT`. I actually think it makes more sense to limit this to `1` than change it here.